src/PureScript/Backend/Chez/Convert.purs

```diff
@@ -213,18 +211,18 @@ codegenPrimOp codegenEnv = case _ of
         OpDivide -> S.Identifier "scm:fl/"
 
       makeComparison :: String -> BackendOperatorOrd -> ChezExpr
-      makeComparison tyName =
+      makeComparison tyName = do
         let
+          comparisonExpression :: String -> ChezExpr
           comparisonExpression opName =
             S.List [ S.Identifier $ Array.fold [ "scm:", tyName, opName ], x', y' ]
-        in
-          case _ of
-            OpEq -> comparisonExpression "=?"
-            OpNotEq -> S.List [ S.Identifier "scm:not", comparisonExpression "=?" ]
-            OpGt -> comparisonExpression ">?"
-            OpGte -> comparisonExpression ">=?"
-            OpLt -> comparisonExpression "<?"
-            OpLte -> comparisonExpression "<=?"
+        case _ of
+          OpEq -> comparisonExpression "=?"
+          OpNotEq -> S.List [ S.Identifier "scm:not", comparisonExpression "=?" ]
+          OpGt -> comparisonExpression ">?"
+          OpGte -> comparisonExpression ">=?"
+          OpLt -> comparisonExpression "<?"
+          OpLte -> comparisonExpression "<=?"
     in
```

Comment:
Another place where 'do...let' can be used?

Reply:
I personally think the `let` block here is too large for do-let. The `in` is a good separator for when you can't see the `let` and vice-versa.